src/Kiota.Builder/Writers/Java/CodeMethodWriter.cs

```diff
@@ -198,77 +211,84 @@ private static void WriteRequestBuilderConstructorCall(CodeMethod codeElement, L
             writer.WriteLine("}");
         }
         private string GetSendRequestMethodName(bool isCollection, string returnType) {
-            if(conventions.PrimitiveTypes.Contains(returnType)) return $"sendPrimitiveAsync";
+            if(conventions.PrimitiveTypes.Contains(returnType)) 
+                if(isCollection)
+                    return $"sendPrimitiveCollectionAsync";
```

Comment:
Unnecessary `$` as not string interpolation is done.

Reply:
```suggestion
                    return "sendPrimitiveCollectionAsync";
```